packages/router-api/src/primitives.rs

```diff
@@ -172,11 +172,24 @@ impl Display for CrossChainId {
     }
 }
 
+/// ChainName represents the identifier used for chains in Amplifier.
+/// Certain restrictions apply on the string:
+/// - Must not be empty
+/// - Must not exceed `ChainName::MAX_LEN` bytes
+/// - Only ASCII characters are allowed
```

Comment:
Prevent utf-8 strings to avoid dealing with edge cases (e.g. lower casing some utf-8 strings can increase their length in bytes)